clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -762,16 +777,33 @@
         RESPONSES[ApiKeys.SYNC_GROUP.id] = SYNC_GROUP_RESPONSE;
         RESPONSES[ApiKeys.DESCRIBE_GROUPS.id] = DESCRIBE_GROUPS_RESPONSE;
         RESPONSES[ApiKeys.LIST_GROUPS.id] = LIST_GROUPS_RESPONSE;
+        RESPONSES[ApiKeys.API_VERSION.id] = API_VERSION_RESPONSE;
 
-        /* set the maximum version of each api */
-        for (ApiKeys api : ApiKeys.values())
+        /* set the minimum and maximum version of each api */
+        for (ApiKeys api : ApiKeys.values()) {
```

Comment:
Gave some more thoughts to it and I think a lot of changes here might collide with KIP-4 changes. If it is OK with you, I would like to do the refactoring in a follow up JIRA, after KIP-4 and KIP-35 changes go in.